ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,158 @@
+    using ..Convex: MOI, solve!, Variable, minimize, evaluate
+    using ..ECOS
+end
+
+# Use other necessary libraries
+using LinearAlgebra: eigvals
+
+# Use functions that are to be extended and additional symbols that are not exported
+using Trixi: Trixi, undo_normalization!, bisect_stability_polynomial, @muladd
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Undo normalization of stability polynomial coefficients by index factorial
+# relative to consistency order.
+function Trixi.undo_normalization!(consistency_order, num_stage_evals, gamma_opt)
+    for k in (consistency_order + 1):num_stage_evals
+        gamma_opt[k - consistency_order] = gamma_opt[k - consistency_order] /
+                                           factorial(k)
+    end
+    return gamma_opt
+end
+
+# Compute stability polynomials for paired explicit Runge-Kutta up to specified consistency
+# order, including contributions from free coefficients for higher orders, and
+# return the maximum absolute value
+function stability_polynomials!(pnoms, consistency_order, num_stage_evals,
+                                normalized_powered_eigvals_scaled,
+                                gamma)
+    num_eig_vals = length(pnoms)
+
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    # First `consistency_order` terms of the exponential
+    for k in 1:consistency_order
+        for i in 1:num_eig_vals
+            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
+        end
+    end
+
+    # Contribution from free coefficients
+    for k in (consistency_order + 1):num_stage_evals
+        pnoms += gamma[k - consistency_order] * normalized_powered_eigvals_scaled[:, k]
+    end
+
+    # For optimization only the maximum is relevant
+    return maximum(abs(pnoms))
+end
+
+#=
+The following structures and methods provide a simplified implementation to 
+discover optimal stability polynomial for a given set of `eig_vals`
+These are designed for the one-step (i.e., Runge-Kutta methods) integration of initial value ordinary 
+and partial differential equations.
+
+- Ketcheson and Ahmadia (2012).
+Optimal stability polynomials for numerical integration of initial value problems
+[DOI: 10.2140/camcos.2012.7.247](https://doi.org/10.2140/camcos.2012.7.247)
+=#
+
+# Perform bisection to optimize timestep for stability of the polynomial
+function Trixi.bisect_stability_polynomial(consistency_order, num_eig_vals,
+                                           num_stage_evals,
+                                           dtmax, dteps, eig_vals,
+                                           verbose = false)
```

Comment:
I strongly recommend to make this a keyword argument
```suggestion
function Trixi.bisect_stability_polynomial(consistency_order, num_eig_vals,
                                           num_stage_evals,
                                           dtmax, dteps, eig_vals;
                                           verbose = false)
```